.changeset/tall-waves-impress.md

```diff
@@ -0,0 +1,58 @@
+---
+'astro': minor
+---
+
+Change Action cookie redirects to an opt-in feature users can implement using middleware. This means Actions submitted from an HTML form action will no longer redirect to the destination as a GET request. Instead, the page will be rendered as a POST result.
+
+This is meant to address the 4 KB size limit users have encountered when calling actions from an HTML form action. It can be difficult to predict the size of an action result for large validation errors or longer return values, like AI chatbot results.
+
+So, we've introduced a new `getActionContext()` utility to let you decide how action results are handled from middleware. You may choose to implement the existing cookie redirect from Astro v4, or implement forwarding with your own session storage.
+
+## Migration
```

Comment:
```suggestion
## Updating your HTML form actions
```
I feel like "migration" maybe sounds a little heavy and daunting here? Maybe something very specific (since this is just about HTML forms, I think? Whatever is appropriate here.) This also lets people know if they're scanning WHAT needs updating, and if they aren't using actions in HTML forms, it's super easy to feel comfortable just moving right along.